platform/ios/src/MGLAnnotationView.mm

```diff
@@ -240,6 +239,7 @@ - (void)setDragState:(MGLAnnotationViewDragState)dragState animated:(BOOL)animat
     }
     else if (dragState == MGLAnnotationViewDragStateCanceling)
     {
+        NSAssert(self.annotation, @"Annotation property should not be nil.");
```

Comment:
When you want to indicate an error on the developer’s part, raise a named NSException instead of calling `NSAssert()`. See https://github.com/mapbox/mapbox-gl-native/pull/8026#discussion_r100661990 for more details. @fabian-guerra, can you change these assertions into named exceptions?